### ISSUE TITLE
Blank lines and comments after null element get lost

### ISSUE THREAD

**Issue by andresmgot** [eemeli/yaml#125]
Hi!

I found a small usability problem when a `null` component gets followed by a comment (related to a different node). The indentation of the comment get lost:

```
> console.log(text)
## Comment 1
foo:

## Comment 2
bar: foo

undefined
> console.log(yaml.parseDocument(text).toString())
## Comment 1
foo:

  null ## Comment 2
bar: foo

undefined
```

The issue get fixed if the `null` key gets comment but I am working with YAML that I cannot control.

Am I missing something?

Thanks in advance! I am happy to work in a PR if this is not an expected behavior.

**Comment by eemeli** [comment]
Unfortunately no, you're not missing anything -- that definitely counts as a bug. PR would be welcome.

**Comment by andresmgot** [comment]
Thanks for the feedback @eemeli. Any pointers about what code should I look for to submit a patch?

**Comment by eemeli** [comment]
Maybe [around here](https://github.com/eemeli/yaml/blob/40acaa76b727b9f3dd028f283e3f10edd8321909/src/schema/parseMap.js#L139-L141)? If it's not exactly that, any fix will most likely touch `parseMap.js` and possibly `parseUtils.js` as well.

**Comment by eemeli** [comment]
This turned out to be related to the same fundamental issue as #126, and solving this actually improved the solution for the other issue as well. I was completely wrong on my earlier guess for the root cause, as it was the CollectionItem CST node parser that was grabbing the comment for the preceding collection value node.

Previously, the collection item parser consumed blank lines and comments until it either found the value, or detected the start of the following node. Now it'll only consume an inline comment, and let the parent handle the rest.

**Comment by andresmgot** [comment]
Awesome. I will test it and let you know if I find any issues. Thank you!

**Comment by eemeli** [comment]
Fix included in patch release 1.7.1.

**Comment by andresmgot** [comment]
:+1: I can confirm that works as expected. The only difference I see now is that some initial spaces are being deleted:

```patch
 ## Add sidecars to the pod.
 ## e.g.
 # - name: your-image-name
-  # image: your-image
-  # imagePullPolicy: Always
-  # ports:
-  # - name: portname
-  #   containerPort: 1234
+# image: your-image
+# imagePullPolicy: Always
+# ports:
+# - name: portname
+#   containerPort: 1234
```

But that's a different/less important issue. Thanks again!
